instrumentation/jmx-metrics/javaagent/camel.md

```diff
@@ -0,0 +1,54 @@
+# Camel Metrics
+
+Here is the list of metrics based on MBeans exposed by Camel.
+
+| Metric Name                                            | Type    | Attributes                             | Description                                                                          |
+|--------------------------------------------------------|---------|----------------------------------------|--------------------------------------------------------------------------------------|
+| org.apache.camel.context.exchanges.completed           | Counter | context, camelVersion                  | Exchanges Completed grouped by Camel Context                                         |
+| org.apache.camel.context.exchanges.failed              | Counter | context, camelVersion                  | Exchanges Failed grouped by Camel Context                                            |
+| org.apache.camel.context.exchanges.inflight            | Counter | context, camelVersion                  | Exchanges Inflight grouped by Camel Context                                          |
+| org.apache.camel.context.exchanges.total               | Counter | context, camelVersion                  | Exchanges Total grouped by Camel Context                                             |
+| org.apache.camel.context.failures.handled              | Counter | context, camelVersion                  | Failures Handled grouped by Camel Context                                            |
+| org.apache.camel.context.external.redeliveries.count   | Counter | context, camelVersion                  | External Redeliveries grouped by Camel Context                                       |
+| org.apache.camel.context.max.processing.time           | Gauge   | context, camelVersion                  | Maximum Processing Time grouped by Camel Context                                     |
+| org.apache.camel.context.mean.processing.time          | Gauge   | context, camelVersion                  | Mean Processing Time grouped by Camel Context                                        |
+| org.apache.camel.context.min.processing.time           | Gauge   | context, camelVersion                  | Min Processing Time grouped by Camel Context                                         |
+| org.apache.camel.context.last.processing.time          | Gauge   | context, camelVersion                  | Last Processing Time grouped by Camel Context                                        |
+| org.apache.camel.context.delta.processing.time         | Gauge   | context, camelVersion                  | Delta Processing Time grouped by Camel Context                                       |
+| org.apache.camel.context.total.processing.time         | Gauge   | context, camelVersion                  | Total Processing Time grouped by Camel Context                                       |
```

Comment:
I think total time should be counter. I have no idea what delta time is so can't tell whether it should be a counter or gauge.

Reply:
Yes, total time definitely is a counter, delta processing time is the difference between processing time of the last two exchange, so a gauge.